drivers/sensor/ti_hdc/ti_hdc.c

```diff
@@ -0,0 +1,209 @@
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct ti_hdc_data *drv_data = dev->driver_data;
+	u64_t tmp;
+
+	/*
+	 * See datasheet "Temperature Register" and "Humidity
+	 * Register" sections for more details on processing
+	 * sample data.
+	 */
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		/* val = -40 + 165 * sample / 2^16 */
+		tmp = 165 * (u64_t)drv_data->t_sample;
+		val->val1 = (s32_t)(tmp >> 16) - 40;
+		val->val2 = (1000000 * (tmp & 0xFFFF)) >> 16;
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP_RAW:
+		val->val1 = drv_data->t_sample;
+		break;
+	case SENSOR_CHAN_HUMIDITY:
+		/* val = 100 * sample / 2^16 */
+		tmp = 100 * (u64_t)drv_data->rh_sample;
+		val->val1 = tmp >> 16;
+		/* x * 1000000 / 65536 == x * 15625 / 1024 */
+		val->val2 = (15625 * (tmp & 0xFFFF)) >> 10;
+		break;
+	case SENSOR_CHAN_HUMIDITY_RAW:
+		val->val1 = drv_data->rh_sample;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api ti_hdc_driver_api = {
+	.sample_fetch = ti_hdc_sample_fetch,
+	.channel_get = ti_hdc_channel_get,
+};
+
+static u16_t read16(struct device *dev, u8_t a, u8_t d)
+{
+	u8_t buf[2];
+
+	if (i2c_burst_read(dev, a, d, (u8_t *)buf, 2) < 0) {
+		LOG_ERR("Error reading register.");
+	}
+	return (buf[0] << 8 | buf[1]);
+}
+
+static int ti_hdc_init(struct device *dev)
+{
+	u16_t helper;
+	struct ti_hdc_data *drv_data = dev->driver_data;
+
+	drv_data->i2c = device_get_binding(DT_TI_HDC_0_BUS_NAME);
+
+	if (drv_data->i2c == NULL) {
+		LOG_DBG("Failed to get pointer to %s device!",
+			DT_TI_HDC_0_BUS_NAME);
+		return -EINVAL;
+	}
+
+	if (read16(drv_data->i2c, DT_TI_HDC_0_BASE_ADDRESS, TI_HDC_MANUFID_REG)
+	    != TI_HDC_MANUF_ID) {
+		LOG_ERR("Failed to get correct manufacturer ID");
+		return -EINVAL;
+	}
+	helper = read16(drv_data->i2c,
+			DT_TI_HDC_0_BASE_ADDRESS,
+			TI_HDC_DEVID_REG);
+	if (helper != TI_HDC1000_DEVID && helper != TI_HDC1050_DEVID) {
+		LOG_ERR("Failed to get correct device ID");
+		return -EINVAL;
+	}
+
+#if defined(CONFIG_TI_HDC_DRDY) && defined(DT_TI_HDC_0_DRDY_GPIOS_CONTROLLER)
+	k_sem_init(&drv_data->data_sem, 0, UINT_MAX);
+
+	/* setup data ready gpio interrupt */
+	drv_data->gpio = device_get_binding(DT_TI_HDC_0_LABEL);
+	if (drv_data->gpio == NULL) {
+		LOG_DBG("Failed to get pointer to %s device",
+			DT_TI_HDC_0_LABEL);
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(drv_data->gpio, DT_TI_HDC_0_DRDY_GPIOS_PIN,
+			   GPIO_DIR_IN | GPIO_INT | GPIO_INT_EDGE |
+#if defined(DT_TI_HDC_GPIO_FLAGS)
+			   DT_TI_HDC_GPIO_FLAGS |
+#endif
+			   GPIO_INT_ACTIVE_LOW | GPIO_INT_DEBOUNCE);
+
+	gpio_init_callback(&drv_data->gpio_cb,
+			   ti_hdc_gpio_callback,
+			   BIT(DT_TI_HDC_0_DRDY_GPIOS_PIN));
+
+	if (gpio_add_callback(drv_data->gpio, &drv_data->gpio_cb) < 0) {
+		LOG_DBG("Failed to set GPIO callback");
+		return -EIO;
+	}
+#endif  /* CONFIG_TI_HDC_DRDY */
+
+	/* configure the sensor */
+	helper = HDC_RH_RES_14BIT | HDC_TEMP_RES_14BIT |
```

Comment:
You're right these are the default values, but this also acts as a demonstration about how someone can change the configuration of the sensor. We can either keep it to this merge request, or I could create a new one after this is merged, to add this only. Whatever suits you best.

Reply:
It does not demonstrate that one can change the configuration. The sensors are almost perfect, after the power on they are just ready and low power per design. I see no need to write the default settings again. It costs time and energy every time the sensor is turned on.

Reply:
The TI offers these configuration options in case someone needs to operate them in a different way. We might not need it now but in the future, someone might need to change it to operate the sensor in a different way.

If the cost is the problem, then I disagree that it affects the power efficiency of any device out there that uses this sensor. It's just a one-off I2C write operation.

Reply:
This commit is not necessary, please drop it. It still not demonstrate that one can change the configuration, It just sets default POR values.